src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -427,6 +429,8 @@ def _parse_entry_func(self):
 
     def _initialize_function(self):
         func = self._parse_entry_func()
+        # persist original function for detailed error message
+        self._original_function = func
```

Comment:
What if the function is already wrapped by @trace?

Reply:
Updated to get from traced function